### DISCUSSION TITLE
Header and footer components

### DISCUSSION THREAD

**Discussion by ianwelerson** [nuxt/nuxt#8021] [General]
Hello people, how it's going?

I have two simple doubts, someone can help me?

First: I'm putting my header and footer on my default layout file, it's okay for an SSR project?

Second: In my nuxt project I have some pages. All pages has the same header and footer, except by one thing on header (an box with text) that I only need in one of them. What is the best approach on that? Create a new layout file with this box and use on this page?

**Comment by cannap** [comment]
create a component and wrap you pages in it so you can tell the component show-box or not 

```
page-with-box.vue

<LayoutWrapper show-box>

</LayoutWrapper>

normal-page.vue
<LayoutWrapper>

</LayoutWrapper>
```
and now you put the Header and Footer in the LayoutWrapper since there is no easy way to pass props into the layout system from nuxt you could use somethlike componentMatched  and check if you have the prop set in the layout system form nuxt but yea 
wrapper is for now the easiest solution
